client/widgetdecorations.cpp

```diff
@@ -179,77 +203,180 @@ void resizable_widget::mousePressEvent(QMouseEvent *event)
+  }
+
+  if (event->x() >= width() / 2 - event_width
+      && event->x() <= width() / 2 + event_width
+      && event->y() >= height() - event_width && event->y() <= height()) {
+    return resizable_type::bottom;
+  }
+
+  if (event->x() >= 0 && event->x() <= event_width
+      && event->y() >= height() - event_width && event->y() <= height()) {
+    return resizable_type::bottomLeft;
+  }
+
+  if (event->x() >= width() - event_width && event->x() <= width()
+      && event->y() >= height() - event_width && event->y() <= height()) {
+    return resizable_type::bottomRight;
+  }
+
+  if (event->x() >= 0 && event->x() <= event_width
+      && event->y() >= height() / 2 - event_width
+      && event->y() <= height() / 2 + event_width) {
+    return resizable_type::left;
+  }
+
+  if (event->x() >= width() - event_width && event->x() <= width()
+      && event->y() >= height() / 2 - event_width
+      && event->y() <= height() / 2 + event_width) {
+    return resizable_type::right;
+  }
+
+  return -1;
+}
+
 /**
    Restores cursor when resizing is done.
  */
 void resizable_widget::mouseReleaseEvent(QMouseEvent *event)
 {
-  QPoint p;
   if (king()->interface_locked) {
     return;
   }
-  if (resize_mode) {
-    resize_mode = false;
-    resx = false;
-    resy = false;
-    resxy = false;
+  if (drag_event != -1) {
+    drag_event = -1;
+    last_position = QPoint{};
     setCursor(Qt::ArrowCursor);
   }
-  p = pos();
   emit resized(rect());
 }
 /**
    Called when mouse moved (mouse track is enabled).
    Used for resizing resizable_widget.
  */
+
 void resizable_widget::mouseMoveEvent(QMouseEvent *event)
 {
   if (king()->interface_locked) {
     return;
   }
-  if ((event->buttons() & Qt::LeftButton) && resize_mode && resy) {
-    QPoint to_move;
-    int newheight = event->globalY() - cursor.y() - geometry().y();
-    resize(width(), this->geometry().height() - newheight);
-    to_move = event->globalPos() - cursor;
-    move(this->x(), to_move.y());
-    setCursor(Qt::SizeVerCursor);
-  } else if (event->x() > width() - 9 && event->y() > 0 && event->y() < 9) {
-    setCursor(Qt::SizeBDiagCursor);
-  } else if ((event->buttons() & Qt::LeftButton) && resize_mode && resx) {
-    resize(event->x(), height());
-    setCursor(Qt::SizeHorCursor);
-  } else if (event->x() > width() - 5 && event->x() < width()) {
-    setCursor(Qt::SizeHorCursor);
-  } else if (event->y() > 0 && event->y() < 5) {
-    setCursor(Qt::SizeVerCursor);
-  } else if (resxy && (event->buttons() & Qt::LeftButton)) {
-    QPoint to_move;
-    int newheight = event->globalY() - cursor.y() - geometry().y();
-    resize(event->x(), this->geometry().height() - newheight);
-    to_move = event->globalPos() - cursor;
-    move(this->x(), to_move.y());
-    setCursor(Qt::SizeBDiagCursor);
+
+  if (event->buttons() & Qt::LeftButton) {
+    if (drag_event != -1) {
+      // constants of min size
+      constexpr auto min_width = 20;
+      constexpr auto min_height = 20;
```

Comment:
Why not `minimumSizeHint`?

Reply:
> Why not `minimumSizeHint`?

`min_width` and `min_height` mean minimum width and height dimensions

Reply:
> Why not `minimumSizeHint`?

thank's, fixed